packages/eslint-plugin/tests/rules/no-unnecessary-type-parameters.test.ts

```diff
@@ -0,0 +1,484 @@
+import { RuleTester } from '@typescript-eslint/rule-tester';
+
+import rule from '../../src/rules/no-unnecessary-type-parameters';
+import { getFixturesRootDir } from '../RuleTester';
+
+const rootPath = getFixturesRootDir();
+
+const ruleTester = new RuleTester({
+  parser: '@typescript-eslint/parser',
+  parserOptions: {
+    sourceType: 'module',
+    tsconfigRootDir: rootPath,
+    project: './tsconfig.json',
+  },
+});
+
+ruleTester.run('no-unnecessary-type-parameters', rule, {
+  valid: [
+    // These tests are adapted from https://effectivetypescript.com/2020/08/12/generics-golden-rule/
```

Comment:
[Praise] These comments were helpful for figuring out coverage of previous rules' logic when reviewing, thanks! 🙂 

...but also, now that we know they're covered and that this rule is going to take over everyone else's, I think it'd be more future-facing to group the test cases by areas of syntax instead. That's what most other tests do in this repo. The "syntax-oriented" style of grouping makes it easier to see what the different test cases are compared to each other. Thoughts?